File: 1848-minimum-distance-to-the-target-element/1848-minimum-distance-to-the-target-element.py
class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        ans = 1_000_000_007
        for i in range(len(nums)):
            if nums[i] == target:
                ans = min(ans, abs(i - start))
        
        return ans

# array
# time: O(n)
# space: O(1)

File: 1848-minimum-distance-to-the-target-element/1848-minimum-distance-to-the-target-element.go
func getMinDistance(nums []int, target int, start int) int {
    ans := 1_000_000_007
    for i := 0; i < len(nums); i++ {
        if nums[i] == target {
            ans = min(ans, abs(i - start))
        }
    }

    return ans
}

func abs(n int) int {
    if n < 0 {
        return -n
    }
    return n
}

// array
// time: O(n)
// space: O(1)